skypy/galaxy/tests/test_redshift.py

```diff
@@ -82,6 +86,26 @@ def test_redshifts_from_comoving_density():
     D, p = kstest(z_gal, lambda z: cosmo.comoving_volume(z).to_value('Mpc3')/V_max)
     assert p > 0.01, 'D = {}, p = {}'.format(D, p)
 
+    # test that error is returned if sky_area is smaller than 0
+    sky_area = -1 * units.steradian
+    with pytest.raises(ValueError):
+        redshifts_from_comoving_density(redshift, density, sky_area, cosmo, noise=False)
+
+    # test that error is returned if sky_area is too big
+    sky_area = 5 * np.pi * units.steradian
+    with pytest.raises(ValueError):
+        redshifts_from_comoving_density(redshift, density, sky_area, cosmo, noise=False)
+
+    # test that error is returned if wrong unit is given
+    sky_area = 1 * units.degree
+    with pytest.raises(UnitsError):
```

Comment:
use `units.UnitsError` instead of importing from the internal `units.core`

```suggestion
    with pytest.raises(units.UnitsError):
```